src/main/resources/vm/script/datamanagement/uploadFunctions.vm

```diff
@@ -203,6 +208,23 @@ function upload {
         then
             uploadGirderFile ${URI}
         fi
+    elif [[ ${URI} == file://* ]]
+    then
+        local FILENAME=`echo $URI | sed 's#file://##g'`
+        local NAME=`basename ${FILENAME}`
+
+        if [ "${TEST}" = "true" ]
+        then
+            FILENAME=${FILENAME}-uploadTest
+            echo "test result" > ${NAME}
+        fi
+
+        \cp $NAME $FILENAME
+
+        if [ "${TEST}" = "true" ]
+        then
+            \rm -f ${NAME}
+        fi
```

Comment:
No need to do the rm here in the test case, as "delete" is called just after when doing the upload test.
Also, I would consider doing a "mv" instead of a "cp".

Reply:
The remove is needed.  The delete is called for the destination file.  Here, we are removing the temporary source file specifically created for testing the upload.

Reply:
mv is now used instead of cp.

Reply:
Small curiosity : what's the use of the backslash before the "mv " and "rm" commands ?
Also, with the mv being used instead of cp, I think the last rm isn't useful.